controllers/journal.js

```diff
@@ -0,0 +1,609 @@
+                });
+            })
+            .catch(err => {
+                res.status(400).send({error: 'List of domains includes invalid domains: ' + err.toString()});
+            });
+    });
+}
+
+exports.addToPublisher = function (req, res) {
+    if (!req.isAuthenticated()) {
+        res.status('401').send();
+        return;
+    }
+
+    if (!req.params.id) {
+        debug('Add Journal to Publisher: No ID provided');
+        res.status(400).send({error: 'No ID provided'});
+        return;
+    }
+
+    if (req.user.level < config.user.level.manage_journal) {
+        debug('[%s] User is not allowed to edit a journal', req.params.id);
+        res.status(401).send();
+        return;
+    }
+
+    Journal.findOne({id: req.params.id}, (err, journal) => {
+        if (err) {
+            res.status('500').send();
+            return;
+        }
+        if (!journal) {
+            res.status('404').send();
+            return;
+        }
+        if (req.user.orcid !== journal.owner) {
+            res.status('403').send();
+            return;
+        }
+        publisher.addJournal(req.body.publisher, journal.id)
+            .then(() => {
+                debug('[%s] Successfully added journal as candidate to publisher %s', journal.id, req.body.publisher);
+                res.status(200).send();
+            })
+            .catch(err => {
+                debug('[%s] Error adding journal as candidate to publisher %s: %O', journal.id, req.body.publisher, err);
+                res.status(400).send({error: 'Error adding journal as candidate to publisher'});
+            });
+    });
+}
+
+exports.listJournal = function (req, res) {
+    debug('Get list of journals');
+    Journal.find({}, '-_id id name domains compendia', (err, journals) => {
+        if (err) {
+            debug('Error getting list of journals from database: %O', err);
+            res.status(500).send("Error getting list of journals from database");
+            return;
+        }
+
+        res.status(200).send(journals);
+    })
+}
+
+exports.viewJournal = function (req, res) {
+    if (!req.isAuthenticated()) {
+        req.status('401').send();
+        return;
+    }
+
+    if (!req.params.id) {
+        res.status('400').send("No ID provided!");
+        return;
+    }
+
+    let journalId = req.params.id;
+
+    debug('[%s] Getting journal', journalId)
+
+    Journal.findOne({id: journalId}, '-_id id name domains compendia', (err, journal) => {
+        if (err) {
+            debug('[%s] Error getting Journal from database: %O', journalId, err);
+            res.status('500').send("Error getting Journal from database");
+            return;
+        } else if (!journal) {
+            debug('[%s] No Journal with this ID found', journalId);
+            res.status('404').send();
+            return;
+        }
+
+        res.status('200').send(journal);
+    });
+}
+
+exports.getJournal = function (req, res) {
```

Comment:
Hier lohnt sich ein kurzer Kommentar im Quelltext - dass es sowohl `viewJournal` als auch `getJournal` gibt finde ich nämlich im ersten Momennt verwirrend. Die Unterscheidung auf Basis des user levels könntest du auch mit einer Funktion machen.
_Sind das auch verschiedene Endpunkte in der API?_